### ISSUE TITLE
Advising methods implemented in an ancestor

### ISSUE THREAD

**Issue by raganwald** [raganwald/YouAreDaChef#4]
if A is an ancestor of B and A.prototype.foo, advising B.foo should not create a naive copy of the method foo in B

**Comment by raganwald** [comment]
Implemented in version 0.0.2!

**Comment by raganwald** [comment]
```
describe 'YouAreDaChef', ->

  it 'shouldn\'t introduce a spurious method', ->

    class Pig extends Animal

    class PotBelliedPig extends Pig

    # we introduce advice in the leaf class
    YouAreDaChef(PotBelliedPig).before 'move', ->
      @name = "A Vietnamese Pot-Bellied Pig named #{@name}"

    # and then we introduce advice in its parent
    YouAreDaChef(Pig).before 'move', ->
      @name = @name.toUpperCase()

    piggie = new Pig('Piggie')

    expect(piggie.move(10)).toBe('PIGGIE moved 10m.')

    porkie = new PotBelliedPig('Porky')

    expect(porkie.move(10)).toBe('A VIETNAMESE POT-BELLIED PIG NAMED PORKY moved 10m.')
```
